[IGNORE][tests] EndToEndTests on helix

radical opened the pull request:
###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/2167)

mitchdenny commented:
Can this PR be closed?

radical committed:
testproject: switch between projectrefs and packagerefs

radical committed:
testproject: skip components

radical committed:
testproject: pkgref/projref - cleanup

radical committed:
dcppack/dashboardpack

radical committed:
workload-testing targets

radical committed:
e2e - deploy, pack archive for tests

radical committed:
yml: install sdk, run with helix

radical committed:
e2e tests

eerhardt commented on tests/testproject/TestProject.AppHost/TestProgram.cs:
```diff
@@ -4,13 +4,29 @@
+        List<string> componentsToSkip = new();
+        for (int i = 0; i < args.Length; i++)
+        {
+            if (args[i].StartsWith("--skip-components", StringComparison.InvariantCultureIgnoreCase))
```
These are called "resources" in the appmodel. We should use that word instead.

eerhardt commented on tests/Aspire.EndToEnd.Tests/IntegrationServicesFixture.cs:
```diff
@@ -127,9 +198,23 @@ private static HttpClient CreateHttpClient()
+                        string msg = $"[{DateTime.Now}] Retry #{args.AttemptNumber+1} for '{args.Outcome.Result?.RequestMessage?.RequestUri}'" +
+                                        $" due to StatusCode: {(int?)args.Outcome.Result?.StatusCode} ReasonPhrase: '{args.Outcome.Result?.ReasonPhrase}'";
+
+                        msg += (args.Outcome.Exception is not null) ? $" Exception: {args.Outcome.Exception.Message} " : "";
```
dump the whole exception tostring.

radical committed:
cleanup

radical committed:
reset workloads.csproj

radical committed:
remove nugetX.config

radical committed:
TestResourceNames

radical committed:
address review feedback

radical committed:
Merge remote-tracking branch 'origin/main' into e2e-testing-combined

radical committed:
cleanup

radical committed:
dcppack.csproj: Lift PackageSource to WorkloadsPackageSource

radical committed:
fix build

radical committed:
debug on ci

radical committed:
dump container list also

radical committed:
disable install-sdk step on windows

radical committed:
trace

radical committed:
helix: use dotnet-latest as payload only if any e2e zip is found

radical committed:
fix resources command line

radical committed:
Fix envvar name

radical committed:
send-to-helix: cleanup

radical committed:
improve workload install for testing

radical committed:
add missing file

radical committed:
fix windows

radical committed:
fix build

radical committed:
Merge remote-tracking branch 'origin/main' into e2e-testing-combined

# Conflicts:
#	tests/testproject/TestProject.IntegrationServiceA/TestProject.IntegrationServiceA.csproj

eerhardt committed:
Remove unnecessary message from backport pipeline (#2474)

JamesNK committed:
Fix opening console logs in a new window or refreshing (#2463)

JamesNK committed:
Fix not found trace message (#2488)

davidfowl committed:
Add support for showing custom resources in the dashboard (#2390)

* Add support for showing custom resources in the dashboard
- Added CustomResourceAnnotation which exposes methods for getting the initial snapshot and changing resource snapshots. These changes show up in the dashboard.
- Show all resources based on the app model before watching dcp for updates. We put a placeholder in the dashboard in the "Starting" state and then remove it when DCP updates come in.

radical committed:
fix vs build

radical committed:
cleanup of messages

radical committed:
show_build_output not needed

radical committed:
Merge remote-tracking branch 'origin/main' into e2e-testing-combined

# Conflicts:
#	playground/CustomResources/CustomResources.AppHost/TestResource.cs
#	src/Aspire.Hosting/ApplicationModel/CustomResourceExtensions.cs
#	src/Aspire.Hosting/Dashboard/ConsoleLogPublisher.cs
#	src/Aspire.Hosting/Dashboard/DcpDataSource.cs

radical committed:
fix local runs

radical committed:
wip

radical committed:
move BE etc into a common folder outside e2e

radical committed:
wip

radical committed:
add missing files

radical committed:
wip

radical committed:
Merge remote-tracking branch 'origin/main' into e2e-testing-combined

# Conflicts:
#	eng/pipelines/templates/BuildAndTest.yml

sebastienros committed:
Use RBAC for Azure AI Search (#2494)

radical committed:
wip

radical committed:
wip

adamint committed:
Dynamically set culture from Accept-Language (#2192)

* Dynamically set culture from Accept-Language

* Add comment on where culture list comes from

* Remove BlazorWebAssemblyLoadAllGlobalizationData property

---------

Co-authored-by: Adam Ratzman <adamratzman@microsoft.com>

radical committed:
add missing file

radical committed:
cleanup

JamesNK committed:
Dashboard correctly logs dynamic port endpoints (#2500)

radical committed:
update nuget.helix.config
